src/integrate/scvi/test.py

```diff
@@ -18,58 +18,95 @@
-                "--modality", "rna",
-                "--obs_batch", "batch",
-                "--output", "output.h5mu",
-                "--model_output", "test/",
-                "--max_epochs", "1",
-                "--n_obs_min_count", "10",
-                "--n_var_min_count", "10",
-                "--output_compression", "gzip"])
-            self.assertTrue(Path("output.h5mu").is_file())
-            output_data = mudata.read_h5mu("output.h5mu")
-            self.assertIn('X_scvi_integrated', output_data.mod['rna'].obsm)
-            self.assertIn('_scvi_batch', output_data.mod['rna'].obs.columns.tolist())
-            self.assertIn('_scvi_labels',  output_data.mod['rna'].obs.columns.tolist())
-            self.assertTrue(Path("test").is_dir())
-            self.assertTrue(Path("test/model.pt").is_file())
-
-
-class TestHVGSubsetting(unittest.TestCase):
-    def test_hvg_subsetting(self):
+def test_scvi(run_component):
+    with NamedTemporaryFile('w', suffix=".h5mu") as tempfile_input_file:
         input_data = mudata.read_h5mu(input_file)
-        adata = input_data.mod["rna"]
+        input_rna = input_data.mod['rna']
+        input_rna.obs['batch'] = 'A'
+        column_index = input_rna.obs.columns.get_indexer(['batch'])
+        input_rna.obs.iloc[slice(input_rna.n_obs//2, None), column_index] = 'B'
+        input_data.write(tempfile_input_file.name)
 
-        old_n_genes = adata.n_vars
+        run_component([
+            "--input", tempfile_input_file.name,
+            "--modality", "rna",
+            "--obs_batch", "batch",
+            "--output", "output.h5mu",
+            "--model_output", "test/",
+            "--max_epochs", "1",
+            "--n_obs_min_count", "10",
+            "--n_var_min_count", "10",
+            "--output_compression", "gzip"])
+        
+        # check files
+        assert Path("output.h5mu").is_file(), "Output file does not exist"
+        assert Path("test").is_dir()
+        assert Path("test/model.pt").is_file()
+
+        # check output h5mu
+        output_data = mudata.read_h5mu("output.h5mu")
+        output_rna = output_data.mod['rna']
+        assert output_rna.n_obs == input_rna.n_obs, f"Number of observations changed\noutput_data: {output_data}"
+        assert output_rna.n_vars == input_rna.n_vars, f"Number of variables changed\noutput_data: {output_data}"
+        assert 'X_scvi_integrated' in output_rna.obsm, f"X_scvi_integrated not added\noutput_data: {output_data}"
 
-        adata.var["highly_variable_features"] = False
-        adata.var.iloc[:old_n_genes // 2, adata.var.columns.get_indexer(["highly_variable_features"])] = True
+        # assert that nothing else has changed
+        del output_rna.obsm["X_scvi_integrated"]
+        assert_equal(input_rna, output_rna)
 
-        adata = subset_vars(adata, subset_col="highly_variable_features")
+def test_scvi_after_hvg(run_component):
+    with NamedTemporaryFile('w', suffix=".h5mu") as tempfile_input_file:
```

Comment:
Could you use the [tmp_path](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture) fixture for this? It provides a unique working directory for each test, so `tempfile_input_file = tmp_path / "input.h5mu"`

Reply:
Makes sense! It's a remnant from the test originally being implemented with `unittest` instead of `pytest` :)